ISeriesProvider/DB2iSeriesTypes.cs

```diff
@@ -1,38 +1,222 @@
 ﻿using System;
+using System.Data;
+using System.Data.Common;
+using System.Reflection;
 
-namespace LinqToDB.DataProvider.DB2iSeries {
```

Comment:
Replaced TypeCreators that doubled as Type holders as TypeDescriptors that contain all the relevant information for each type along with the creation logic. This is much cleaner in usage and also lifecycle